frontend/src/app/patients/ArchivePersonModal.tsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import { gql, useMutation } from "@apollo/client";
+import Modal from "react-modal";
+import Button from "../commonComponents/Button";
+import { displayFullName, showNotification } from "../utils";
+import "./ArchivePersonModal.scss";
+
+import Alert from "../commonComponents/Alert";
+import { toast } from "react-toastify";
+
+const ARCHIVE_PERSON_RECORD = gql`
+  mutation ArchivePerson($id: String!, $deleted: Boolean!) {
+    setPatientIsDeleted(id: $id, deleted: $deleted) {
+      internalId
+    }
+  }
+`;
+
+interface Props {
+  person: any;
+  closeModal: () => void;
+}
+
+const ArchivePersonModal = ({ person, closeModal }: Props) => {
+  const [archivePersonRecord] = useMutation(ARCHIVE_PERSON_RECORD);
```

Comment:
This can be typed with:
```suggestion
interface ArchivePersonResponse {
  internalId: string;
}
interface ArchivePersonParams {
  id: string,
  deleted: string
}
const ArchivePersonModal = ({ person, closeModal }: Props) => {
  const [archivePersonRecord] = useMutation<ArchivePersonResponse, ArchivePersonParams>(ARCHIVE_PERSON_RECORD);
```